app/master/build.py

```diff
@@ -112,10 +112,13 @@ def generate_project_type(self):
         project_type_params = self.build_request.build_parameters()
         project_type_params.update({'build_project_directory': build_specific_project_directory})
         self._project_type = util.create_project_type(project_type_params)
-
         if self._project_type is None:
             raise BuildProjectError('Build failed due to an invalid project type.')
 
+        # Create the build results directory.
+        if not os.path.exists(self._build_results_dir()):
+            os.mkdir(self._build_results_dir())
```

Comment:
I think there is a helper to create a directory (`app.util.fs.create_dir`) that be a bit cleaner?

Also, I'm curious about your thoughts behind putting result dir creation in `generate_project_type()` (vs. e.g., `prepare()`).